src/indexable.cr

```diff
@@ -872,6 +879,13 @@ module Indexable(T)
     nil
   end
 
+  # :ditto:
+  #
+  # Raises `Enumerable::NotFoundError` if no match is found.
+  def rindex!(offset = size - 1, & : T ->)
+    rindex(offset) { |i| yield i } || raise Enumerable::NotFoundError.new
```

Comment:
The parameter of the block is the element itself, so `e` is a better name than `i` (see `#index!`'s implementation)